core/src/main/java/bisq/core/account/sign/SignedWitnessService.java

```diff
@@ -75,10 +76,15 @@
     private final ArbitratorManager arbitratorManager;
     private final User user;
 
-    @Getter
     private final Map<P2PDataStorage.ByteArray, SignedWitness> signedWitnessMap = new HashMap<>();
+
+    // The getSignedWitnessSet is called very often and is a bit expensive. We cache the result in that map but we
+    // remove the cache entry if we get a matching SignedWitness added to the signedWitnessMap.
+    private final Map<P2PDataStorage.ByteArray, Set<SignedWitness>> getSignedWitnessSetCache = new ConcurrentHashMap<>();
```

Comment:
The name is a bit off for a map, just `signedWitnessSetCache` would be more natural.

Reply:
Agree...